common/util/http_util.go

```diff
@@ -253,3 +259,34 @@ func CheckConnect(ip string, port int, timeout int) (localIP string, e error) {
 	}
 	return
 }
+
+// IsExpired checks if a resource received or stored is the same.
+func IsExpired(url string, headers map[string]string, lastModified int64, eTag string) (bool, error) {
+	if lastModified <= 0 && IsEmptyStr(eTag) {
+		return true, nil
+	}
+
+	// set headers
+	if headers == nil {
+		headers = make(map[string]string)
+	}
+	if lastModified > 0 {
+		headers["If-Modified-Since"] = strconv.FormatInt(lastModified, 10)
+	}
+	if !IsEmptyStr(eTag) {
+		headers["If-None-Match"] = eTag
+	}
+
+	// send request
+	resp, err := HTTPWithHeaders("HEAD", url, headers)
+	if err != nil {
+		return true, err
```

Comment:
If `HEAD` is failed, then there is a high probability that `GET` will failed. At this moment, we may use the cached data rather than downloading it again.

Reply:
Has been updated. PTAL.